src/main/java/org/jenkinsci/plugins/gitclient/JGitAPIImpl.java

```diff
@@ -1473,7 +1477,9 @@ else if (!referencePath.isDirectory())
                             .setCredentialsProvider(getProvider())
                             .setTagOpt(tags ? TagOpt.FETCH_TAGS : TagOpt.NO_TAGS)
                             .setRefSpecs(refspecs);
-                    if (timeout != null) fetch.setTimeout(timeout);
+                    if (timeout != null) {
+                        fetch.setTimeout(timeout * 60);
```

Comment:
Unfortunately, this won't work in a way that is consistent with the timeout implementation in the command line git implementation.

Command line git timeout sets the maximum duration of a single git command.

JGit FetchCommand.setTimeout() sets the maximum period during which there was not communication on the JGit transport.  Refer to comments on [my failed attempt](https://github.com/MarkEWaite/git-client-plugin/commit/ecaf1b2bad99fcfb6c4b44e0374513fd27d279f2) to add timeout using this technique.

Reply:
I remembered a comment from you a while back about this but was not aware of the referenced commit.

Reply:
I had this small change locally already prepared some months ago as you can see by the commit dates and just wanted to get it out the door. Unfortunately I would not want to spent much more time on this. So it would also be ok for me that we close the pull request unmerged.

Reply:
Yes, I was so happy to discover the setTimeout() API on JGit.  It was only when writing several tests that should have failed that I discovered that the JGit concept of timeout is well suited to an API while the git plugin concept of timeout is well suited to an end user.

Reply:
@darxriggs I think we should close this pull request unless you plan to implement the JGit timeout with separate threads or some other technique outside the JGit timeout method.

What do you think?